backend/src/main/java/gov/cdc/usds/simplereport/service/DbOrgRoleClaimsService.java

```diff
@@ -0,0 +1,147 @@
+      return List.of(getOrganizationRoleClaims(user));
+    } catch (NonexistentUserException | MisconfiguredUserException e) {
+      return new ArrayList<>();
+    }
+  }
+
+  public OrganizationRoleClaims getOrganizationRoleClaims(ApiUser user) {
+    Set<Organization> orgs = user.getOrganizations();
+    if (orgs.size() != 1) {
+      log.error("Misconfigured organizations in DB for User ID: {}", user.getInternalId());
+      throw new MisconfiguredUserException();
+    }
+    Set<OrganizationRole> roles = user.getRoles();
+    Set<UUID> facilityUUIDs =
+        user.getFacilities().stream().map(Facility::getInternalId).collect(Collectors.toSet());
+
+    String orgExternalId = orgs.stream().findFirst().get().getExternalId();
+    return new OrganizationRoleClaims(orgExternalId, facilityUUIDs, roles);
+  }
+
+  /**
+   * Compares two lists of OrganizationRoleClaims and checks if they are equal; If they are not
+   * equal, a message is logged with the affected User ID
+   *
+   * @param oktaClaims - List<OrganizationRoleClaims> from Okta
+   * @param dbClaims - List<OrganizationRoleClaims> from DB
+   * @return boolean
+   */
+  public boolean checkOrgRoleClaimsEquality(
+      List<OrganizationRoleClaims> oktaClaims, List<OrganizationRoleClaims> dbClaims) {
+    boolean hasEqualRoleClaims = false;
+    if (oktaClaims.size() == dbClaims.size()) {
+      List<OrganizationRoleClaims> sanitizedOktaClaims = sanitizeOktaOrgRoleClaims(oktaClaims);
+      hasEqualRoleClaims =
+          sanitizedOktaClaims.stream()
+              .allMatch(
+                  sanitizedOktaClaim ->
+                      dbClaims.stream()
+                          .anyMatch(dbClaim -> equalOrgRoleClaim(sanitizedOktaClaim, dbClaim)));
+    }
+    if (!hasEqualRoleClaims) {
+      logUnequalClaims();
+    }
+
+    return hasEqualRoleClaims;
+  }
+
+  /** Logs a message saying OrganizationRoleClaims are unequal with the affected User ID */
+  private void logUnequalClaims() {
+    // WIP: Currently assumes check is for the current user
+    // This may change based on where checkOrgRoleClaimsEquality is called
+    String username = _getCurrentUser.get().getUsername();
+    ApiUser user = _userRepo.findByLoginEmail(username).orElseThrow(NonexistentUserException::new);
+    log.error(
+        "Okta OrganizationRoleClaims do not match database OrganizationRoleClaims for User ID: {}",
+        user.getInternalId());
+  }
+
+  /**
+   * Removes NO_ACCESS OrganizationRole in order to compare with OrganizationRole from DB, NO_ACCESS
+   * role does not exist in DB, only in Okta
+   *
+   * @param oktaClaims - List<OrganizationRoleClaims> from Okta
+   * @return list of OrganizationRoleClaims without NO_ACCESS OrganizationRole
+   */
+  private List<OrganizationRoleClaims> sanitizeOktaOrgRoleClaims(
+      List<OrganizationRoleClaims> oktaClaims) {
+    return oktaClaims.stream()
+        .map(
+            oktaClaim -> {
```

Comment:
I think we can simplify this using `.filter`

```
    private List<OrganizationRoleClaims> sanitizeOktaOrgRoleClaims(
            List<OrganizationRoleClaims> oktaClaims) {
        return oktaClaims.stream()
                .map(
                        oktaClaim -> {
                            Set<OrganizationRole> orgRoles = oktaClaim.getGrantedRoles().stream()
                                    .filter(c -> c != OrganizationRole.NO_ACCESS)
                                    .collect(Collectors.toSet());
                            return new OrganizationRoleClaims(
                                    oktaClaim.getOrganizationExternalId(),
                                    oktaClaim.getFacilities(),
                                    orgRoles);
                        }
                )
                .collect(Collectors.toList());
    }
```

Reply:
Ah! Thank you! 🧹

Reply:
Done!